// let promise1 = new Promise((resolve, reject) => {
//     let result = true;
//     if (result){
//         resolve(resolved);

//     }  else{
//         reject(rejected);
//     }
//     })

//     console.log(promise1);

    // let promise = new promise ((resolve, reject) =>{
        
    // })
    // promise.then(result) =>{
    //     console.log("res");
    // }.catch((error) =>{
    //     console.log
    // })

    // function fetchdata (){
    //     return new promise ((resolve, reject) =>{
    //         setTimeout(() => {
    //             let data = "data fetched"
    //             resolve(data)
    //         }, 2000)
    //     })
    // }
    // fetchdata().then((result) =>{
    //     console.log("fetching the data");

    // })
    // .then((result) =>{
    //     setTimeout(() => {
    //         console.log("data is received");
    //     }, 2000);
    // })
    // .then((result) =>{
    //     setTimeout(() => {
    //         console.log("data is processed");
    //     }, 3000);
    // })
    // .then((result) =>{
    //     setTimeout(() => {
    //         console.log("data is completed");
    //     }, 2000);
    // })
    // .catch((error) =>{
    //     console.log("error is occured");
    // })

//    const resolvedpromise = Promise.resolve("already resolved");
//    console.log(resolvedpromise); 

// promises methods 
// promise.all

const promise1 = promise.resolve("first");
const promise2 = new promise((resolve) =>{

})

